.github/workflows/main.yml

```diff
@@ -0,0 +1,46 @@
+name: tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  changed-files:
+    runs-on: ubuntu-latest
+    outputs:
+      any_nr_changed: ${{ steps.changed-files.outputs.any_changed }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: tj-actions/changed-files@v44
+        id: changed-files
+        with:
+          files: |
+            packages/**/*.{nr,toml}
+            Nargo.toml
+
+  style:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: yarn
+      - run: yarn
+      - run: yarn check
+
+  test:
+    needs: changed-files
+    if: needs.changed-files.outputs.any_nr_changed == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: noir-lang/noirup@v0.1.3
+        with:
+          toolchain: 0.26.0
+      - run: nargo test
```

Comment:
Here the steps to install node and the node deps were removed. As they aren't necessary. We only need the nargo bin to run the tests with `nargo test`